flang/lib/Lower/OpenMP.cpp

```diff
@@ -178,14 +179,27 @@ genOMP(Fortran::lower::AbstractConverter &converter,
                          &clause.u)) {
         const Fortran::parser::OmpObjectList &ompObjectList = copyinClause->v;
         genObjectList(ompObjectList, converter, copyinClauseOperands);
-      }
+      } else if(const auto &allocateClause = 
+                  std::get_if<Fortran::parser::OmpAllocateClause>(&clause.u)){
+        if (const auto &allocatorValue = 
+              std::get<std::optional<Fortran::parser::OmpAllocateClause::Allocator>>(allocateClause->t)){
+          allocator = fir::getBase(converter.genExprValue(*Fortran::semantics::GetExpr(allocatorValue->v)));
```

Comment:
I think `allocator` is not used after being pushed in the vector. Would it make sense to declare a simple Value here in this scope?

Reply:
I'll try and do that